operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -811,6 +811,10 @@ void addStartupEnvVars(List<V1EnvVar> vars) {
     addEnvVar(vars, ServerEnvVars.AS_SERVICE_NAME, LegalNames.toServerServiceName(getDomainUid(), getAsName()));
     Optional.ofNullable(getDataHome()).ifPresent(v -> addEnvVar(vars, ServerEnvVars.DATA_HOME, v));
     addEnvVarIfTrue(mockWls(), vars, "MOCK_WLS");
+
+    if (getDomainHomeSourceType() == DomainSourceType.FromModel) {
+      addEnvVar(vars, ServerEnvVars.FAIL_BOOT_ON_SITUATIONAL_CONFIG_ERROR, "false");
```

Comment:
This is going to change our pod hash (`weblogic.sha256`) which means that any MII pod that is running when the operator is upgraded will be rolled. Can this instead be set in the pod start script (`startServer.sh`).

Reply:
Just to clarify the scenario you're describing for my understanding, this is when upgrading the Operator from a previous MII domain that uses sit config overrides to this version which uses the model filter?  The one place that uses this environment is in startNodeManager.sh (as a server startup argument) and I couldn't determine how we know the domain source type is 'FromModel' (MII).  serverStart.sh does:

if [ -f /weblogic-operator/introspector/domainzip.secure ]; then
...
fi

I think this should work.  Let me make the change and verify.

Reply:
Right, a customer is using operator 3.1.4 and has a MII domain. The admin server and some number of managed servers are running. The customer upgrades to 3.2.0 containing your work. I would expect that pods do not roll simply because I upgrade the operator.

Thinking more deeply about this... The change to add the extra environment variable would have caused the existing pods to roll. I think that we have a workaround in that we can set this in startServer.sh. However, we likely have an earlier issue... Your change to not generate config overrides will mean that the generated domain home will be different. This change would also cause existing pods to roll, correct? If so, we should talk together with @jshum2479 and see if we can come up with a way to roll this out so that existing pods are not rolled or restarted, but newly created pods come up with the new pattern.

Reply:
We can add logic in start script to reset the env when the DOMAIN_SOURCE_TYPE is FromModel.

Reply:
It doesn't look like it'd hurt to leave it alone. It'd become a NO-OP.

Reply:
The `FAIL_BOOT_ON_SITUATIONAL_CONFIG_ERROR` is for the core config override logic to failed to start the server if the override file is wrong, the start server script already have the logic to set it to true if the environment is not set,  the env is for the case if user choose to override it.  

I am not seeing a reason to set it for mii since we are using override and it will cause the roll because of the pod spec hash changed because the newly introduced environment variable.

Reply:
The overall point is that there's no need to _change_ the current setting in this pull.  We don't care what it's set to in the new path...

Reply:
in startNodeManager.sh, we set FAIL_BOOT_ON_SITUATIONAL_CONFIG_ERROR system property to 'true' by default.  Managed servers fail to start/boot with the following error:

<Mar 29, 2021 8:22:23 PM GMT> <Critical> <Management> <BEA-141327> <The SitConfigRequired configuration element is set to true, but no situational config files were found.> 
<Mar 29, 2021 8:22:23 PM GMT> <Critical> <WebLogicServer> <BEA-000362> <Server failed. Reason: 141327> 
<Mar 29, 2021 8:22:23 PM GMT> <Notice> <WebLogicServer> <BEA-000365> <Server state changed to FAILED.> 

The validation is here:
if (isSitConfigRequired() && unexpiredFiles.isEmpty()) {
        if (debugLogger.isDebugEnabled()) {
          debugLogger.debug( "[SitConfig] isSitConfigRequired is true and no Sit Config Files exist. Failing boot.");
        }
        String l = ManagementLogger.logSituationalConfigRequired();
        throw new ServiceFailureException(l);
}

So we should either set it to 'false' or not define it for MII.

Reply:
The point is that we're not supporting custom or internal sit-cfg in MII regardless, so there won't be any sit-cfg that can fail, so the flag should have no effect. If removing it is safe (doesn't cause a roll), then feel free to remove it. Or feel free to  leave it untouched.